sync.go

```diff
@@ -42,6 +42,10 @@ var (
 	disabled = func(args ...interface{}) string { return fmt.Sprint(args...) }
 )
 
+// Running too many syncs at once can exhaust file descriptor limits.
+// Empirically, ~90 is enough to hit the macOS default limit of 256.
+const maxConcurrentSyncs = 10
```

Comment:
Most of the time is spent blocked on I/O. I don't think the number of cpus is important here?